### PR TITLE
+act Add Runnable support for ReceiveBuilder's `matchEquals` and `mat…

### PR DESCRIPTION
**Pull request by He-Pin**
refs:https://github.com/akka/akka/issues/25872

### CODE REVIEW

File: akka-actor-tests/src/test/java/akka/japi/pf/ReceiveBuilderTest.java [hunk 1]
```diff
@@ -209,10 +209,15 @@ public void shouldMatchByExternalPredicate(){
...
     Msg2 msg2 = new Msg2("foo");
+    final FI.TypedPredicate<Integer> greaterThanOrEqual18 = anyInt -> anyInt >= 18;
+    final FI.TypedPredicate<Integer> greaterThanOrEqual16 = anyInt -> anyInt >= 16;
     Receive rcv = ReceiveBuilder.create()
         .matchEquals(msg2, m -> result("match msg2"))
         .matchEquals("foo", m -> result("match foo"))
+        .matchEquals("bar", () -> result("match bar"))
         .matchEquals(17, m -> result("match 17"))
+        .matchEquals(20,greaterThanOrEqual18,m -> result("match 20"))
+        .matchEquals(16,greaterThanOrEqual16,() -> result("match 16"))
```
Comment (hepin1989) [inline akka-actor-tests/src/test/java/akka/japi/pf/ReceiveBuilderTest.java]: Add some tests which are missing before .

**Comment by hepin1989** [comment]
```
Error Message
akka.remote.RemoteTransportException: Inbound Aeron channel is in errored state. See Aeron logs for details.
Stacktrace
sbt.ForkMain$ForkError: akka.remote.RemoteTransportException: Inbound Aeron channel is in errored state. See Aeron logs for details.
	at akka.remote.artery.aeron.ArteryAeronUdpTransport.retry$1(ArteryAeronUdpTransport.scala:236)
	at akka.remote.artery.aeron.ArteryAeronUdpTransport.blockUntilChannelActive(ArteryAeronUdpTransport.scala:227)
	at akka.remote.artery.aeron.ArteryAeronUdpTransport.runInboundStreams(ArteryAeronUdpTransport.scala:306)
	at akka.remote.artery.ArteryTransport.start(ArteryTransport.scala:456)
	at akka.remote.RemoteActorRefProvider.init(RemoteActorRefProvider.scala:222)
	at akka.actor.ActorSystemImpl.liftedTree2$1(ActorSystem.scala:873)
	at akka.actor.ActorSystemImpl._start$lzycompute(ActorSystem.scala:869)
	at akka.actor.ActorSystemImpl._start(ActorSystem.scala:869)
	at akka.actor.ActorSystemImpl.start(ActorSystem.scala:890)
	at akka.actor.ActorSystem$.apply(ActorSystem.scala:246)
	at akka.actor.ActorSystem$.apply(ActorSystem.scala:289)
	at akka.actor.ActorSystem$.apply(ActorSystem.scala:264)
	at akka.testkit.AkkaSpec.<init>(AkkaSpec.scala:67)
	at akka.stream.scaladsl.StreamRefsSpec.<init>(StreamRefsSpec.scala:186)
	at akka.stream.scaladsl.StreamRefsSpec.<init>(StreamRefsSpec.scala:190)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:422)
	at java.lang.Class.newInstance(Class.java:442)
	at org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:435)
	at sbt.ForkMain$Run.lambda$runTest$1(ForkMain.java:304)
	at sbt.ForkMain$Run$$Lambda$280/1334405552.call(Unknown Source)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by hepin1989** [comment]
need https://github.com/akka/akka/pull/25822 to get merged first.

**Comment by hepin1989** [comment]
The error is unrelated, create https://github.com/akka/akka/issues/26009

File: akka-actor-tests/src/test/java/akka/japi/pf/ReceiveBuilderTest.java (hunk 1 above)
Comment (patriknw) [reply akka-actor-tests/src/test/java/akka/japi/pf/ReceiveBuilderTest.java]: one thing that could be interesting to look into is how understandable compilation errors become when we add a lot of overloads

I'm not fully convinced that this convenience is worth adding. You can use `__` as variable name if you don't care about it

Comment (hepin1989) [reply akka-actor-tests/src/test/java/akka/japi/pf/ReceiveBuilderTest.java]: hm,then both are two keypress `__` and `()`

Comment (hepin1989) [reply akka-actor-tests/src/test/java/akka/japi/pf/ReceiveBuilderTest.java]: But from a Java user perspective, If I wrote a `__` in my code, my workmate will say `What!?`

Comment (patriknw) [reply akka-actor-tests/src/test/java/akka/japi/pf/ReceiveBuilderTest.java]: yeah, learning something new is terrible

(joking aside, I completely understand)

Comment (hepin1989) [reply akka-actor-tests/src/test/java/akka/japi/pf/ReceiveBuilderTest.java]: Yes, My life will be much easier if my work currently is writing Scala, but for maintenance reason, we are using only Java now. But thanks god, we could use Java 8, but still much developer here don't want to get out the Java 6 pool.

**Comment by hepin1989** [comment]
Close this cause I think it's ok to write the `__` or `msg` now.

**Pull request closed without merging**